examples/cc2538dk/crypto/sha256-test.c

```diff
@@ -0,0 +1,264 @@
+        NULL,
+        NULL
+      }, {
+        0x15, 0xd2, 0x3e, 0xea, 0x57, 0xb3, 0xd4, 0x61,
+        0xbf, 0x38, 0x91, 0x12, 0xab, 0x4c, 0x43, 0xce,
+        0x85, 0xe1, 0x68, 0x23, 0x8a, 0xaa, 0x54, 0x8e,
+        0xc8, 0x6f, 0x0c, 0x9d, 0x65, 0xf9, 0xb9, 0x23
+      }
+    }, { /* Message of length 128 */
+      {
+        "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklabcd"
+        "efghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijkl",
+        NULL,
+        NULL
+      }, {
+        0xf8, 0xa3, 0xf2, 0x26, 0xfc, 0x42, 0x10, 0xe9,
+        0x0d, 0x13, 0x0c, 0x7f, 0x41, 0xf2, 0xbe, 0x66,
+        0x45, 0x53, 0x85, 0xd2, 0x92, 0x0a, 0xda, 0x78,
+        0x15, 0xf8, 0xf7, 0x95, 0xd9, 0x44, 0x90, 0x5f
+      }
+    }, { /* Message of length 64 */
+      {
+        "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijkl",
+        NULL,
+        NULL
+      }, {
+        0x2f, 0xcd, 0x5a, 0x0d, 0x60, 0xe4, 0xc9, 0x41,
+        0x38, 0x1f, 0xcc, 0x4e, 0x00, 0xa4, 0xbf, 0x8b,
+        0xe4, 0x22, 0xc3, 0xdd, 0xfa, 0xfb, 0x93, 0xc8,
+        0x09, 0xe8, 0xd1, 0xe2, 0xbf, 0xff, 0xae, 0x8e
+      }
+    }, { /* Message of length 66 */
+      {
+        "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmn",
+        NULL,
+        NULL
+      }, {
+        0x92, 0x90, 0x1c, 0x85, 0x82, 0xe3, 0x1c, 0x05,
+        0x69, 0xb5, 0x36, 0x26, 0x9c, 0xe2, 0x2c, 0xc8,
+        0x30, 0x8b, 0xa4, 0x17, 0xab, 0x36, 0xc1, 0xbb,
+        0xaf, 0x08, 0x4f, 0xf5, 0x8b, 0x18, 0xdc, 0x6a
+      }
+    }, {
+      {
+        "abcdbcdecdefde",
+        "fgefghfghighijhijkijkljklmklmnlmnomnopnopq",
+        NULL
+      }, {
+        0x24, 0x8d, 0x6a, 0x61, 0xd2, 0x06, 0x38, 0xb8,
+        0xe5, 0xc0, 0x26, 0x93, 0x0c, 0x3e, 0x60, 0x39,
+        0xa3, 0x3c, 0xe4, 0x59, 0x64, 0xff, 0x21, 0x67,
+        0xf6, 0xec, 0xed, 0xd4, 0x19, 0xdb, 0x06, 0xc1
+      }
+    }, {
+      {
+        "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijkl",
+        "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijkl",
+        NULL
+      }, {
+        0xf8, 0xa3, 0xf2, 0x26, 0xfc, 0x42, 0x10, 0xe9,
+        0x0d, 0x13, 0x0c, 0x7f, 0x41, 0xf2, 0xbe, 0x66,
+        0x45, 0x53, 0x85, 0xd2, 0x92, 0x0a, 0xda, 0x78,
+        0x15, 0xf8, 0xf7, 0x95, 0xd9, 0x44, 0x90, 0x5f
+      }
+    }, {
+      {
+        "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefgh",
+        "ijkl",
+        "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijkl"
+      }, {
+        0xf8, 0xa3, 0xf2, 0x26, 0xfc, 0x42, 0x10, 0xe9,
+        0x0d, 0x13, 0x0c, 0x7f, 0x41, 0xf2, 0xbe, 0x66,
+        0x45, 0x53, 0x85, 0xd2, 0x92, 0x0a, 0xda, 0x78,
+        0x15, 0xf8, 0xf7, 0x95, 0xd9, 0x44, 0x90, 0x5f
+      }
+    }
+  };
+  static sha256_state_t state;
+  static int i, j;
+  static uint8_t ret;
+  static rtimer_clock_t total_time;
+  rtimer_clock_t time;
+  size_t len;
+  uint8_t sha256[32];
```

Comment:
Correct. That might be it, and it should anyway be fixed. Does it work if you make it static?